sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -971,9 +971,38 @@ object TypeCoercion {
         case (ArrayType(fromType, true), ArrayType(toType: DataType, false)) => null
 
         case (ArrayType(fromType, false), ArrayType(toType: DataType, false))
-            if !Cast.forceNullable(fromType, toType) =>
+          if !Cast.forceNullable(fromType, toType) =>
           implicitCast(fromType, toType).map(ArrayType(_, false)).orNull
 
+        // Implicit cast between Map types.
+        // Follows the same semantics of implicit casting between two array types.
+        // Refer to documentation above.
+        case (MapType(fromKeyType, fromValueType, fn), MapType(toKeyType, toValueType, true))
+          if !Cast.forceNullable(fromKeyType, toKeyType) =>
+          val newKeyType = implicitCast(fromKeyType, toKeyType).orNull
+          val newValueType = implicitCast(fromValueType, toValueType).orNull
+          if (newKeyType != null && newValueType != null
+            && (!newKeyType.sameType(fromKeyType) || !newValueType.sameType(fromValueType))) {
```

Comment:
Do we need this?

Reply:
@ueshin U r right. Let me take it out and run the tests.